client/src/main/java/com/vesoft/nebula/client/graph/data/Relationship.java

```diff
@@ -25,18 +27,38 @@ public Relationship(Edge edge) {
         this.edge = edge;
     }
 
-    public ValueWrapper srcId() throws UnsupportedEncodingException {
-        return new ValueWrapper(edge.src, decodeType);
+    /**
+     * get the src id from the relationship
+     * @return ValueWrapper, if int id, you can call srcId().asLong(),
+     *     if string id, you can call srcId().asString()
+     */
+    public ValueWrapper srcId() {
+        return edge.type > 0 ? new ValueWrapper(edge.src, decodeType)
+            : new ValueWrapper(edge.dst, decodeType);
```

Comment:
Why edge.dst is the srcId when edge.type >0?

For edge `a->b`, its srcId is a.
For edge `a<-b`, its srcId is b.

Reply:
> Why edge.dst is the srcId when edge.type >0?
edge.dst is the srcId when edge.type <0